.travis.yml

```diff
@@ -1,20 +1,34 @@
 language: go
 
+env:
+  global:
+    - DOCKER_IMAGE_NAME=$TRAVIS_REPO_SLUG
+    - DOCKER_IMAGE_TAG=$TRAVIS_BRANCH
+    - DOCKER_IMAGE_TAG_COMMIT=${TRAVIS_COMMIT::8}
+
+services:
+  - docker
+
 go:
-  - 1.11
-  - tip
+  - 1.12
 
 go_import_path: github.com/turbonomic/kubeturbo
 
 before_install:
   - go get -v github.com/mattn/goveralls
 
-matrix:
-  allow_failures:
-    - go: tip
-
 script:
   - make fmtcheck 
   - make vet
-  - make product
-  - if [ "$TRAVIS_GO_VERSION" == "tip" ] ; then make test ; else $HOME/gopath/bin/goveralls -v -race -service=travis-ci ; fi
+  - make build
+  - $HOME/gopath/bin/goveralls -v -race -service=travis-ci
+  - cp ./kubeturbo build
+  - cd build
+  - docker build -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG --build-arg GIT_COMMIT=$TRAVIS_COMMIT --label "git-version=$TRAVIS_COMMIT" .
```

Comment:
We do need to do:
  - docker pull registry.access.redhat.com/ubi8
before the build in order to pull the latest ubi8, which will update the base image with the security patch from RH

Reply:
@enlinxu I have verified that every travis build will boot a new virtual machine to create a new build environment, and there will be no existing docker images. That means the base image will always be pulled from redhat registry as seen from the build log:

![image](https://user-images.githubusercontent.com/10012486/60842353-ef69b680-a1a1-11e9-93de-f462c845b8e9.png)